ui/page/send/page.go

```diff
@@ -121,19 +136,56 @@ func NewSendPage(l *load.Load) *Page {
 	pg.feeRateSelector.ContainerInset = layout.Inset{Bottom: values.MarginPadding100}
 	pg.feeRateSelector.WrapperInset = layout.UniformInset(values.MarginPadding15)
 
+	pg.initializeAccountSelectors()
+
+	pg.sendDestination.addressChanged = func() {
+		pg.validateAndConstructTx()
+	}
+
+	pg.amount.amountChanged = func() {
+		pg.validateAndConstructTxAmountOnly()
+	}
+
+	pg.initLayoutWidgets()
+
+	return pg
+}
+
+// initWalletSelector is used for the send modal to for wallet selection.
+func (pg *Page) initWalletSelector() {
+	// initialize wallet selector
+	pg.sourceWalletSelector = components.NewWalletAndAccountSelector(pg.Load).
+		Title(values.String(values.StrSelectWallet))
+	pg.selectedWallet = pg.sourceWalletSelector.SelectedWallet()
+
+	// Source wallet picker
+	pg.sourceWalletSelector.WalletSelected(func(selectedWallet *load.WalletMapping) {
+		pg.selectedWallet = selectedWallet
+		pg.amount.setAssetType(selectedWallet.GetAssetType())
+		pg.sendDestination.initDestinationWalletSelector(selectedWallet.GetAssetType())
+		pg.initializeAccountSelectors()
+		pg.resetDestinationAccountSelectors()
+	})
+}
+
+func (pg *Page) resetDestinationAccountSelectors() {
+	// this resets the selected destination account based on the
+	// selected source account. This is done to prevent sending to
+	// an account that is invalid either because the destination
+	// account is the same as the source account or because the
+	// destination account needs to change based on if the selected
+	// wallet has privacy enabled.
+	pg.sendDestination.destinationAccountSelector.SelectFirstValidAccount(
+		pg.sendDestination.destinationWalletSelector.SelectedWallet())
+	pg.validateAndConstructTx()
+}
+
+func (pg *Page) initializeAccountSelectors() {
 	// Source account picker
-	pg.sourceAccountSelector = components.NewWalletAndAccountSelector(l).
+	pg.sourceAccountSelector = components.NewWalletAndAccountSelector(pg.Load).
 		Title(values.String(values.StrFrom)).
 		AccountSelected(func(selectedAccount *sharedW.Account) {
-			// this resets the selected destination account based on the
-			// selected source account. This is done to prevent sending to
-			// an account that is invalid either because the destination
-			// account is the same as the source account or because the
-			// destination account needs to change based on if the selected
-			// wallet has privacy enabled.
-			pg.sendDestination.destinationAccountSelector.SelectFirstValidAccount(
-				pg.sendDestination.destinationWalletSelector.SelectedWallet())
-			pg.validateAndConstructTx()
+			pg.resetDestinationAccountSelectors()
```

Comment:
I think the comment is still very much relevant here. It basically explains why you're calling a destination account reset method when the source account is changed.

Also, should `pg.resetDestinationAccountSelectors()` be `pg.resetDestinationAccountSelector()`. Are there multiple destination account selectors?